File: collegeresults/WebContent/scripts/addresult.js
$(document).ready(init);

function init(){
	$("#sessionaltype").css("display","none");

	$("input[name='exam']").click(function(){
		var value = $(this).val();
		
		if(value=="Sessional")
			$("#sessionaltype").css("display","block");
		else
			$("#sessionaltype").css("display","none");
	});
}

function checkform(){
	
	var rollno = $("input[name='txtrollno']").val();
	var subject = $("input[name='txtsubject']").val();
	var maximum = $("input[name='txtmaximum']").val();
	var minimum = $("input[name='txtminimum']").val();
	var obtained = $("input[name='txtobtained']").val();
	
	var valid = true;
	
	$(".msg").html("");
	
	if(rollno.length==0){
		valid = false;
		$("#msgrollno").html("?");
	}
	
	if(subject.length==0){
		valid = false;
		$("#msgsubject").html("?");
	}
	
	if(maximum.length==0 || isNaN(maximum)){
		valid = false;
		$("#msgmarks").html("Invalid maximum marks!");
	}
	else if(minimum.length==0 || isNaN(minimum)){
		valid = false;
		$("#msgmarks").html("Invalid minimum marks!");
	}	
	else if(obtained.length==0 || isNaN(obtained)){
		valid = false;
		$("#msgmarks").html("Invalid obtained marks!");
	}
	
	return valid;
}

File: collegeresults/src/com/results/servlets/AddResult.java
package com.results.servlets;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import pojo.Marks;

import com.results.beans.MarksBean;
import com.results.beans.StudentBean;

@WebServlet("/addresult")
public class AddResult extends HttpServlet {
	private static final long serialVersionUID = 1L;

	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

		String rollno = request.getParameter("txtrollno");
		String year = request.getParameter("year");
		String semester = request.getParameter("semester");
		String exam = request.getParameter("exam");
		String sessionalType = request.getParameter("sessionaltype");
		String maximum = request.getParameter("txtmaximum");
		String minimum = request.getParameter("txtminimum");
		String obtained = request.getParameter("txtobtained");
		String subject = request.getParameter("txtsubject");
		
		Marks marks = new Marks();
		
		marks.setRollno(rollno);
		marks.setExam(exam);
		marks.setYear(year);
		marks.setSemester(semester);
		marks.setSubject(subject);
		marks.setSessionalType(sessionalType);
		marks.setMaximum(Integer.parseInt(maximum));
		marks.setMinimum(Integer.parseInt(minimum));
		marks.setObtained(Integer.parseInt(obtained));
		
		MarksBean bean = new MarksBean();
		StudentBean sbean = new StudentBean();
		
		sbean.setRollno(rollno);
		
		if(sbean.getStudentByRollno()==null)
			response.sendRedirect("addresult.jsp?msg=Invalid roll number");
		else{
			String added = bean.addMarks(marks);
			
			if(added.equals("added"))
				response.sendRedirect("addresult.jsp?msg=Marks Added");
			else if(added.equals("failed"))
				response.sendRedirect("addresult.jsp?msg=Failed");
			else if(added.equals("error"))
				response.sendRedirect("addresult.jsp?msg=Error");
		}
	}
}
